include/cudaq/Optimizer/Transforms/Passes.td

```diff
@@ -141,6 +141,40 @@ def CheckKernelCalls : Pass<"check-kernel-calls", "mlir::func::FuncOp"> {
   }];
 }
 
+def ClassicalOptimization : Pass<"classical-optimization"> {
+  let summary = "Perform classical optimizations until a fix point is reached.";
+  let description = [{
+    Performs a number of classical optimizations greedily until a fix point
+    is reached:
+    - canonicalization
+    - simplify regions
+    - write-after-write-elimination
+    - lift-array-alloc
+    - cc-loop-unroll
+
+    The signal-failure-if-any-loop-cannot-be-completely-unrolled option controls
+    whether to signal a failure if all loops cannot be fully unrolled. This is
+    necessary when synthesizing quantum circuits from CUDA-Q kernels, such
+    as when generating a QIR base profile. A quantum circuit requires all loops
+    be completely unrolled.
+  }];
+
+  let dependentDialects = ["mlir::arith::ArithDialect",
+                           "mlir::cf::ControlFlowDialect"];
+
+  let options = [
+    Option<"threshold", "maximum-iterations", "unsigned", /*default=*/"50",
+      "Maximum iterations to unroll.">,
+    Option<"signalFailure",
+      "signal-failure-if-any-loop-cannot-be-completely-unrolled", "bool",
+      /*default=*/"false", "Signal failure if pass can't unroll all loops.">,
```

Comment:
Do we need this? The fix-point may be before every single cc.loop is unrolled. (This option isn't composable.)